engine/annotations/openpose.py

```diff
@@ -0,0 +1,255 @@
+                    (0, 1, 0, 0),
+                    (0, 0, -1, 0),
+                    (0, 0, 0, 1)
+                ))
+                gpu.matrix.load_matrix(mathutils.Matrix.Identity(4))
+                gpu.matrix.load_projection_matrix(projection_matrix)
+                gpu.state.blend_set('ALPHA')
+
+                def transform(x, y):
+                    return (
+                        (x - 0.5) * 2 * ratio,
+                        (y - 0.5) * 2
+                    )
+
+                shader = gpu.shader.from_builtin('3D_UNIFORM_COLOR')
+                batch = batch_for_shader(shader, 'TRI_STRIP', {"pos": [(-ratio, -1, 0), (-ratio, 1, 0), (ratio, -1, 0), (ratio, 1, 0)]})
+                shader.uniform_float("color", (0, 0, 0, 1))
+                batch.draw(shader)
+
+                for object in (context.scene.objects if collection is None else collection.objects):
+                    object = object.evaluated_get(context)
+                    if object.hide_render:
+                        continue
+                    if object.pose is None:
+                        continue
+                    for connection, color in lines.items():
+                        a, a_side = connection[0].identify(object.data, object.pose)
+                        b, b_side = connection[1].identify(object.data, object.pose)
+                        if a is None or b is None:
+                            continue
+                        a = bpy_extras.object_utils.world_to_camera_view(context.scene, context.scene.camera, object.matrix_world @ (a.tail if a_side == Side.TAIL else a.head))
+                        b = bpy_extras.object_utils.world_to_camera_view(context.scene, context.scene.camera, object.matrix_world @ (b.tail if b_side == Side.TAIL else b.head))
+                        draw_ellipse_2d(transform(a[0], a[1]), transform(b[0], b[1]), .015, 32, (color[0] / 255.0, color[1] / 255.0, color[2] / 255.0, 0.5))
+                    for b in Bone:
+                        bone, side = b.identify(object.data, object.pose)
+                        color = b.color()
+                        if bone is None: continue
+                        tail = bpy_extras.object_utils.world_to_camera_view(context.scene, context.scene.camera, object.matrix_world @ (bone.tail if side == Side.TAIL else bone.head))
+                        draw_circle_2d(transform(tail[0], tail[1]), .015, 16, (color[0] / 255.0, color[1] / 255.0, color[2] / 255.0, 0.5))
+
+            depth = np.array(fb.read_color(0, 0, width, height, 4, 0, 'FLOAT').to_list())
+        offscreen.free()
+        result = depth
+        e.set()
+    bpy.app.timers.register(_execute, first_interval=0)
+    e.wait()
+    return result
+
+def draw_circle_2d(center, radius, segments, color):
+    m = (1.0 / (segments - 1)) * (math.pi * 2)
+
+    coords = [
+        (
+            center[0] + math.cos(m * p) * radius,
+            center[1] + math.sin(m * p) * radius
```

Comment:
Seems to require the z axis instead of inferring it as 0
```suggestion
            center[0] + math.cos(m * p) * radius,
            center[1] + math.sin(m * p) * radius,
            0
```